cmd/process-agent/main_common.go

```diff
@@ -350,3 +218,138 @@ func initRuntimeSettings() {
 		}
 	}
 }
+
+type miscDeps struct {
+	fx.In
+	Lc fx.Lifecycle
+
+	Config   config.Component
+	Syscfg   sysprobeconfig.Component
+	HostInfo hostinfo.Component
+}
+
+// initMisc initializes modules that cannot, or have not yet been componetized.
+// Todo: (Components) WorkloadMeta, remoteTagger, telemetry Server, expvars, api server
+func initMisc(deps miscDeps) error {
+	initRuntimeSettings()
+
+	if err := statsd.Configure(ddconfig.GetBindHost(), deps.Config.GetInt("dogstatsd_port")); err != nil {
+		_ = log.Criticalf("Error configuring statsd: %s", err)
+		return err
+	}
+
+	if err := ddutil.SetupCoreDump(deps.Config); err != nil {
+		_ = log.Warnf("Can't setup core dumps: %v, core dumps might not be available after a crash", err)
+	}
+
+	// Setup workloadmeta
+	var workloadmetaCollectors workloadmeta.CollectorCatalog
+	if deps.Config.GetBool("process_config.remote_workloadmeta") {
+		workloadmetaCollectors = workloadmeta.RemoteCatalog
+	} else {
+		workloadmetaCollectors = workloadmeta.NodeAgentCatalog
+	}
+	store := workloadmeta.CreateGlobalStore(workloadmetaCollectors)
+
+	// Setup remote tagger
+	var t tagger.Tagger
+	if deps.Config.GetBool("process_config.remote_tagger") {
+		options, err := remote.NodeAgentOptions()
+		if err != nil {
+			_ = log.Errorf("unable to deps.Configure the remote tagger: %s", err)
+		} else {
+			t = remote.NewTagger(options)
+		}
+	} else {
+		t = local.NewTagger(store)
+	}
+	tagger.SetDefaultTagger(t)
+
+	// Run a profile & telemetry server.
+	if deps.Config.GetBool("telemetry.enabled") {
+		http.Handle("/telemetry", telemetry.Handler())
+	}
+
+	expvarPort := getExpvarPort(deps.Config)
+	expvarServer := &http.Server{Addr: fmt.Sprintf("localhost:%d", expvarPort), Handler: http.DefaultServeMux}
+
+	// Initialize status
+	err := initStatus(deps.HostInfo.Object(), deps.Syscfg)
+	if err != nil {
+		log.Critical("Failed to initialize status:", err)
+	}
+
+	deps.Lc.Append(fx.Hook{
+		OnStart: func(ctx context.Context) error {
+			store.Start(ctx)
+
+			err := tagger.Init(ctx)
+			if err != nil {
+				_ = log.Errorf("failed to start the tagger: %s", err)
+			}
+
+			go func() {
+				_ = expvarServer.ListenAndServe()
+			}()
+
+			// Run API server
+			err = api.StartServer()
+			if err != nil {
+				_ = log.Error(err)
+			}
+
+			err = manager.ConfigureAutoExit(ctx, deps.Config)
```

Comment:
<!-- Warning represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**🔨 warning:** Are you sure want to pass `ctx` as first parameter to `ConfigureAutoExit(...)` call ?  Because this context provided by fx only live for a given time (15 secs by default). In this case it would have the effect to stop *autoexit* within 15 seconds.

Here‏ is the fx specification about `onStart` context:
```
		// To mitigate the impact of deadlocks in application startup and
		// shutdown, Fx imposes a time limit on OnStart and OnStop hooks. By
		// default, hooks have a total of 15 seconds to complete. Timeouts are
		// passed via Go's usual context.Context.
```
link: https://pkg.go.dev/go.uber.org/fx#section-readme